### PR TITLE
[IDP-465] feat(refactor): .NET 6 for subscriber project

### PR DESCRIPTION
**Pull request by felpel**
## Core changes
- Convert the subscriber project to target the .NET 6 framework rather than .NET Standard 2.0
- Add an extension method to register the event grid APIs through dependency injection

### CODE REVIEW

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    /// </remarks>
+    public class SubscriptionValidationEventTestData
+    {
+        public string ValidationCode { get; init; }
+    }
+
+    [Fact]
+    public async Task GivenEventsApi_WhenASubscriptionEventIsPosted_ThenReturnsOkWithValidationResponse()
+    {
+        A.CallTo(() => this._topicProvider.GetTopicValidationPattern(A<string>._)).Returns("dummytopic");
```
Comment (Gcaya) [inline src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiIntegrationTests.cs:47]: We could add some Give When Then comments in the tests 😁

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        A.CallTo(() => this._eventGridRequestHandler.HandleRequestAsync(A<object>._, A<CancellationToken>._, requestTelemetry)).Returns(Task.FromResult(eventGridRequestResult));
+
+        var actualResult = await EventsApi.HandleEventGridEvent(new object(), this._httpContext, this._eventGridRequestHandler, CancellationToken.None);
+
+        var (data, statusCode) = await actualResult.GetResponseAsync<SubscriptionValidationResponse>();
+
+        Assert.Equal(HttpStatusCode.OK, statusCode);
+        Assert.NotNull(data);
+        Assert.Equal(subscriptionValidationResponse.ValidationResponse, data.ValidationResponse);
```
Comment (Gcaya) [inline src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs:51]:
```suggestion
        Assert.Equal(subscriptionValidationResponse.ValidationResponse, data.ValidationResponse);
        A.CallTo(() => this._eventGridRequestHandler.HandleRequestAsync(A<object>._, A<CancellationToken>._, requestTelemetry)).MustHaveHappenedOnceExactly();
```

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        };
+
+        A.CallTo(() => this._eventGridRequestHandler.HandleRequestAsync(A<object>._, A<CancellationToken>._, requestTelemetry)).Returns(Task.FromResult(eventGridRequestResult));
+
+        var actualResult = await EventsApi.HandleEventGridEvent(new object(), this._httpContext, this._eventGridRequestHandler, CancellationToken.None);
+
+        var (data, statusCode) = await actualResult.GetResponseAsync<object>();
+
+        Assert.Equal(HttpStatusCode.OK, statusCode);
+        Assert.Null(data);
```
Comment (Gcaya) [inline src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs:75]:
```suggestion
        Assert.Null(data);
        A.CallTo(() => this._eventGridRequestHandler.HandleRequestAsync(A<object>._, A<CancellationToken>._, requestTelemetry)).MustHaveHappenedOnceExactly();
```

**Review by Gcaya — APPROVED** [review]

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiIntegrationTests.cs (hunk 1 above)
Comment (felpel) [reply src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiIntegrationTests.cs:51]: # :memo: ✅

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs (hunk 2 above)
Comment (felpel) [reply src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs:54]: As discussed, given the mocks are strict, we can assume that this call will happen once.

File: src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs (hunk 3 above)
Comment (felpel) [reply src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiUnitTests.cs:81]: Idem as the previous comment

**Review by Gcaya — APPROVED** [review]

**Pull request merged**